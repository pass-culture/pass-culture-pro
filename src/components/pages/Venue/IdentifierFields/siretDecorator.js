import createDecorator from 'final-form-calculate'

import { getSireInfo, SIRET } from 'utils/sire'

const emptyValues = {
  address: null,
  city: null,
  latitude: null,
  longitude: null,
  name: null,
  postalCode: null,
}

export const createSiretDecorator = createDecorator({
  field: 'siret',
  updates: async siret => {
    const siretInfo = await getSireInfo(siret, SIRET)
    if (!siretInfo) return {}
    if (siretInfo.error) return emptyValues
    return siretInfo.values
  },
})

export default createSiretDecorator
