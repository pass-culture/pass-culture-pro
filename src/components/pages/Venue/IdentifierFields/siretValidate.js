import { getSireInfo, SIRET } from 'utils/siren'

export const siretValidate = async siret => {
  const siretInfo = await getSireInfo(siret, SIRET)
  if (!siretInfo || !siretInfo.error) return undefined
  return siretInfo.error
}

export default siretValidate
