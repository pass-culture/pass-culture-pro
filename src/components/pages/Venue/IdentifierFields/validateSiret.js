import { getSireInfo, SIRET } from 'utils/sire'

export const validateSiret = async siret => {
  const siretInfo = await getSireInfo(siret, SIRET)
  if (!siretInfo || !siretInfo.error) return undefined
  return siretInfo.error
}

export default validateSiret
