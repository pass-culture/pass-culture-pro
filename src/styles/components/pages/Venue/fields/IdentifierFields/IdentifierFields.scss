.siret-label-details {
  font-size: 0.8125rem;
}

.siret-label-bold {
  font-weight: bold;
}
