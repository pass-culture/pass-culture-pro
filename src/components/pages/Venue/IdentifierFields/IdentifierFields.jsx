import PropTypes from 'prop-types'
import React from 'react'

import siretValidate from './siretValidate'
import { Icon } from 'components/layout/Icon'
import {
  HiddenField,
  TextareaField,
  TextField,
} from 'components/layout/form/fields'
import { formatSirenOrSiret } from 'utils/siren'

const getIsCommentRequired = formSiret => !formSiret || formSiret.length !== 14

const buildSiretLabel = isCreatedEntity =>
  `SIRET${isCreatedEntity ? ' (si applicable)' : ''} : `

const IdentifierFields = ({
  fieldReadOnlyBecauseFrozenFormSiret,
  formSiret,
  initialSiret,
  isCreatedEntity,
  isModifiedEntity,
  readOnly,
}) => (
  <div className="section">
    <h2 className="main-list-title is-relative">
      IDENTIFIANTS
      {!readOnly && (
        <span className="is-pulled-right is-size-7 has-text-grey">
          Les champs marqués d'un <span className="required-legend"> * </span>{' '}
          sont obligatoires
        </span>
      )}
    </h2>
    <div className="field-group">
      {isCreatedEntity && <HiddenField name="managingOffererId" />}
      <TextField
        format={formatSirenOrSiret}
        innerClassName="col-50"
        label={buildSiretLabel(isCreatedEntity)}
        name="siret"
        readOnly={readOnly || initialSiret !== null}
        renderValue={() => {
          if (readOnly) {
            return null
          }
          if (fieldReadOnlyBecauseFrozenFormSiret) {
            return (
              <span
                className="button"
                data-place="bottom"
                data-tip="<p>Il n'est pas possible de modifier le nom, l'addresse et la géolocalisation du lieu quand un siret est renseigné.</p>"
                data-type="info">
                <Icon svg="picto-info" />
              </span>
            )
          }
          return (
            <span
              className="button"
              data-place="bottom"
              data-tip="<div><p>Saisissez ici le SIRET du lieu lié à votre structure pour retrouver ses informations automatiquement.</p>
<p>Si les informations ne correspondent pas au SIRET saisi, <a href='mailto:pass@culture.gouv.fr?subject=Question%20SIRET'> contactez notre équipe</a>.</p></div>"
              data-type="info">
              <Icon svg="picto-info" />
            </span>
          )
        }}
        type="siret"
        {...(initialSiret ? {} : { validate: siretValidate })}
      />
      <TextField
        label="Nom : "
        name="name"
        readOnly={readOnly || fieldReadOnlyBecauseFrozenFormSiret}
        required
      />
      <TextField label="Nom d'usage : " name="publicName" readOnly={readOnly} />
      <TextField
        innerClassName="col-75"
        label="E-mail : "
        name="bookingEmail"
        readOnly={readOnly}
        required
        renderValue={() => {
          if (readOnly) {
            return null
          }
          return (
            <span
              className="button"
              data-tip="<p>Cette adresse recevra les e-mails de notification de réservation (sauf si une adresse différente est saisie lors de la création d'une offre)</p>"
              data-place="bottom"
              data-type="info">
              <Icon svg="picto-info" />
            </span>
          )
        }}
        type="email"
      />
      <TextareaField
        label="Commentaire (si pas de SIRET) : "
        name="comment"
        readOnly={readOnly}
        required={getIsCommentRequired(formSiret)}
        rows={1}
      />
    </div>
  </div>
)

IdentifierFields.defaultProps = {
  fieldReadOnlyBecauseFrozenFormSiret: false,
  formSiret: null,
  initialSiret: null,
  readOnly: true,
}

IdentifierFields.propTypes = {
  fieldReadOnlyBecauseFrozenFormSiret: PropTypes.bool,
  formSiret: PropTypes.string,
  initialSiret: PropTypes.string,
  isCreatedEntity: PropTypes.bool.isRequired,
  isModifiedEntity: PropTypes.bool.isRequired,
  readOnly: PropTypes.bool,
}

export default IdentifierFields
