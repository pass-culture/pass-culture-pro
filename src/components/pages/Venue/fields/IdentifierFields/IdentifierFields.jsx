import PropTypes from 'prop-types'
import React, { Fragment, PureComponent } from 'react'

import siretValidate from './validators/siretValidate'
import Icon from '../../../../layout/Icon'
import HiddenField from '../../../../layout/form/fields/HiddenField'
import TextareaField from '../../../../layout/form/fields/TextareaField'
import TextField from '../../../../layout/form/fields/TextField'
import { formatSiret } from '../../siret/formatSiret'
import ReactTooltip from 'react-tooltip'


class IdentifierFields extends PureComponent {
  componentDidUpdate() {
    ReactTooltip.rebuild()
  }

  handleRenderValue = (fieldReadOnlyBecauseFrozenFormSiret, readOnly) => () => {
    if (readOnly) {
      return null
    }
    if (fieldReadOnlyBecauseFrozenFormSiret) {
      return (
        <span
          className="button"
          data-place="bottom"
          data-tip="<p>Il n’est pas possible de modifier le nom, l’addresse et la géolocalisation du lieu quand un siret est renseigné.</p>"
          data-type="info"
        >
          <Icon svg="picto-info" />
        </span>
      )
    }
    return (
      <span
        className="button"
        data-place="bottom"
        data-tip="<div><p>Saisissez ici le SIRET du lieu lié à votre structure pour retrouver ses informations automatiquement.</p>
        <p>Si les informations ne correspondent pas au SIRET saisi, <a href='mailto:pass@culture.gouv.fr?subject=Question%20SIRET'> contactez notre équipe</a>.</p></div>"
        data-type="info"
      >
        <Icon svg="picto-info" />
      </span>
    )
  }

  handleRender = readOnly => () => {
    if (readOnly) {
      return null
    }
    return (
      <span
        className="button"
        data-place="bottom"
        data-tip="<p>Cette adresse recevra les e-mails de notification de réservation (sauf si une adresse différente est saisie lors de la création d’une offre)</p>"
        data-type="info"
      >
        <Icon svg="picto-info" />
      </span>
    )
  }

  commentValidate = comment => {
    const {
      formSiret,
    } = this.props

    if (formSiret && formSiret.length === 14) {
      return ''
    }
    if (comment === undefined || comment === '') {
      return 'Ce champ est obligatoire'
    }
    return ''
  }

  render() {
    const {
      fieldReadOnlyBecauseFrozenFormSiret,
      initialSiret,
      isCreatedEntity,
      readOnly,
    } = this.props

    return (
      <div className="section">
        <h2 className="main-list-title is-relative">
          {'INFORMATIONS DU LIEU'}
          {!readOnly && (
            <span className="is-pulled-right fs13 has-text-grey">
              {'Les champs marqués d’un'}
              <span className="required-legend">
                {' * '}
              </span>
              {' sont obligatoires'}
            </span>
          )}
        </h2>
        <div className="field-group">
          {isCreatedEntity && <HiddenField name="managingOffererId" />}
          <div className="field text-field is-label-aligned">
            <label
              className="field-label"
              htmlFor="siret"
            >
              {isCreatedEntity ? (
                <Fragment>
                  {'SIRET'}
                  <span className="siret-label-details">
                    <span className="siret-label-bold">
                      {' du lieu qui accueille vos offres'}
                    </span>
                    {' (si applicable) : '}
                  </span>
                </Fragment>
              ) : (
                'SIRET : '
              )}
            </label>
            <TextField
              format={formatSiret}
              name="siret"
              readOnly={readOnly || initialSiret !== null}
              renderValue={this.handleRenderValue(fieldReadOnlyBecauseFrozenFormSiret, readOnly)}
              type="siret"
              {...(initialSiret ? {} : { validate: siretValidate })}
            />
          </div>
          <TextField
            label="Nom du lieu : "
            name="name"
            readOnly={readOnly || fieldReadOnlyBecauseFrozenFormSiret}
            required
          />
          <TextField
            label="Nom d’usage du lieu : "
            name="publicName"
            readOnly={readOnly}
          />
          <TextField
            innerClassName="col-75"
            label="E-mail : "
            name="bookingEmail"
            readOnly={readOnly}
            renderValue={this.handleRender(readOnly)}
            required
            type="email"
          />
          <TextareaField
            label="Commentaire (si pas de SIRET) : "
            name="comment"
            readOnly={readOnly}
            rows={1}
            validate={this.commentValidate}
          />
        </div>
      </div>
    )
  }
}

IdentifierFields.defaultProps = {
  fieldReadOnlyBecauseFrozenFormSiret: false,
  formSiret: null,
  initialSiret: null,
  readOnly: true,
}

IdentifierFields.propTypes = {
  fieldReadOnlyBecauseFrozenFormSiret: PropTypes.bool,
  formSiret: PropTypes.string,
  initialSiret: PropTypes.string,
  isCreatedEntity: PropTypes.bool.isRequired,
  readOnly: PropTypes.bool,
}

export default IdentifierFields
