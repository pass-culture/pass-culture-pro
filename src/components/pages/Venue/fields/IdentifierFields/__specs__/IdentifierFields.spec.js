import { shallow } from 'enzyme'
import React from 'react'
import IdentifierFields from '../IdentifierFields'

describe('src | components | pages | Venue | fields | IdentifierFields', () => {
  let props

  beforeEach(() => {
    props = {
      fieldReadOnlyBecauseFrozenFormSiret: true,
      formSiret: 'form siret',
      initialSiret: 'form siret',
      isCreatedEntity: true,
      isModifiedEntity: true,
      readOnly: true
    }
  })

  describe('snapshot', () => {
    it('should match snapshot', () => {
      // when
      const wrapper = shallow(<IdentifierFields {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
