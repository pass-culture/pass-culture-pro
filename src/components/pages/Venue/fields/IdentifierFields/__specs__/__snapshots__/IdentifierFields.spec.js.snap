// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Venue | fields | IdentifierFields snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdentifierFields
    fieldReadOnlyBecauseFrozenFormSiret={true}
    formSiret="form siret"
    initialSiret="form siret"
    isCreatedEntity={true}
    isModifiedEntity={true}
    readOnly={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="main-list-title is-relative"
        >
          IDENTIFIANTS
        </h2>,
        <div
          className="field-group"
        >
          <Unknown
            name="managingOffererId"
            validator={[Function]}
          />
          <TextField
            className=""
            disabled={false}
            format={[Function]}
            id={null}
            innerClassName="col-50"
            label="SIRET (si applicable) : "
            name="siret"
            parse={null}
            placeholder=""
            readOnly={true}
            renderInner={[Function]}
            renderValue={[Function]}
            required={false}
            type="siret"
            validate={null}
          />
          <TextField
            className=""
            disabled={false}
            format={null}
            id={null}
            innerClassName={null}
            label="Nom : "
            name="name"
            parse={null}
            placeholder=""
            readOnly={true}
            renderInner={[Function]}
            renderValue={[Function]}
            required={true}
            type="text"
            validate={null}
          />
          <TextField
            className=""
            disabled={false}
            format={null}
            id={null}
            innerClassName={null}
            label="Nom d'usage : "
            name="publicName"
            parse={null}
            placeholder=""
            readOnly={true}
            renderInner={[Function]}
            renderValue={[Function]}
            required={false}
            type="text"
            validate={null}
          />
          <TextField
            className=""
            disabled={false}
            format={null}
            id={null}
            innerClassName="col-75"
            label="E-mail : "
            name="bookingEmail"
            parse={null}
            placeholder=""
            readOnly={true}
            renderInner={[Function]}
            renderValue={[Function]}
            required={true}
            type="email"
            validate={null}
          />
          <TextareaField
            autoComplete={false}
            className=""
            disabled={false}
            label="Commentaire (si pas de SIRET) : "
            maxLength={1000}
            name="comment"
            placeholder=""
            readOnly={true}
            required={true}
            rows={1}
            validate={null}
            validating={false}
          />
        </div>,
      ],
      "className": "section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "IDENTIFIANTS",
            false,
          ],
          "className": "main-list-title is-relative",
        },
        "ref": null,
        "rendered": Array [
          "IDENTIFIANTS",
          false,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              name="managingOffererId"
              validator={[Function]}
            />,
            <TextField
              className=""
              disabled={false}
              format={[Function]}
              id={null}
              innerClassName="col-50"
              label="SIRET (si applicable) : "
              name="siret"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={false}
              type="siret"
              validate={null}
            />,
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName={null}
              label="Nom : "
              name="name"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="text"
              validate={null}
            />,
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName={null}
              label="Nom d'usage : "
              name="publicName"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={false}
              type="text"
              validate={null}
            />,
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName="col-75"
              label="E-mail : "
              name="bookingEmail"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="email"
              validate={null}
            />,
            <TextareaField
              autoComplete={false}
              className=""
              disabled={false}
              label="Commentaire (si pas de SIRET) : "
              maxLength={1000}
              name="comment"
              placeholder=""
              readOnly={true}
              required={true}
              rows={1}
              validate={null}
              validating={false}
            />,
          ],
          "className": "field-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "managingOffererId",
              "validator": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "disabled": false,
              "format": [Function],
              "id": null,
              "innerClassName": "col-50",
              "label": "SIRET (si applicable) : ",
              "name": "siret",
              "parse": null,
              "placeholder": "",
              "readOnly": true,
              "renderInner": [Function],
              "renderValue": [Function],
              "required": false,
              "type": "siret",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "disabled": false,
              "format": null,
              "id": null,
              "innerClassName": null,
              "label": "Nom : ",
              "name": "name",
              "parse": null,
              "placeholder": "",
              "readOnly": true,
              "renderInner": [Function],
              "renderValue": [Function],
              "required": true,
              "type": "text",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "disabled": false,
              "format": null,
              "id": null,
              "innerClassName": null,
              "label": "Nom d'usage : ",
              "name": "publicName",
              "parse": null,
              "placeholder": "",
              "readOnly": true,
              "renderInner": [Function],
              "renderValue": [Function],
              "required": false,
              "type": "text",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "disabled": false,
              "format": null,
              "id": null,
              "innerClassName": "col-75",
              "label": "E-mail : ",
              "name": "bookingEmail",
              "parse": null,
              "placeholder": "",
              "readOnly": true,
              "renderInner": [Function],
              "renderValue": [Function],
              "required": true,
              "type": "email",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "autoComplete": false,
              "className": "",
              "disabled": false,
              "label": "Commentaire (si pas de SIRET) : ",
              "maxLength": 1000,
              "name": "comment",
              "placeholder": "",
              "readOnly": true,
              "required": true,
              "rows": 1,
              "validate": null,
              "validating": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="main-list-title is-relative"
          >
            IDENTIFIANTS
          </h2>,
          <div
            className="field-group"
          >
            <Unknown
              name="managingOffererId"
              validator={[Function]}
            />
            <TextField
              className=""
              disabled={false}
              format={[Function]}
              id={null}
              innerClassName="col-50"
              label="SIRET (si applicable) : "
              name="siret"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={false}
              type="siret"
              validate={null}
            />
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName={null}
              label="Nom : "
              name="name"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="text"
              validate={null}
            />
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName={null}
              label="Nom d'usage : "
              name="publicName"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={false}
              type="text"
              validate={null}
            />
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName="col-75"
              label="E-mail : "
              name="bookingEmail"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="email"
              validate={null}
            />
            <TextareaField
              autoComplete={false}
              className=""
              disabled={false}
              label="Commentaire (si pas de SIRET) : "
              maxLength={1000}
              name="comment"
              placeholder=""
              readOnly={true}
              required={true}
              rows={1}
              validate={null}
              validating={false}
            />
          </div>,
        ],
        "className": "section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "IDENTIFIANTS",
              false,
            ],
            "className": "main-list-title is-relative",
          },
          "ref": null,
          "rendered": Array [
            "IDENTIFIANTS",
            false,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                name="managingOffererId"
                validator={[Function]}
              />,
              <TextField
                className=""
                disabled={false}
                format={[Function]}
                id={null}
                innerClassName="col-50"
                label="SIRET (si applicable) : "
                name="siret"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={false}
                type="siret"
                validate={null}
              />,
              <TextField
                className=""
                disabled={false}
                format={null}
                id={null}
                innerClassName={null}
                label="Nom : "
                name="name"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={true}
                type="text"
                validate={null}
              />,
              <TextField
                className=""
                disabled={false}
                format={null}
                id={null}
                innerClassName={null}
                label="Nom d'usage : "
                name="publicName"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={false}
                type="text"
                validate={null}
              />,
              <TextField
                className=""
                disabled={false}
                format={null}
                id={null}
                innerClassName="col-75"
                label="E-mail : "
                name="bookingEmail"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={true}
                type="email"
                validate={null}
              />,
              <TextareaField
                autoComplete={false}
                className=""
                disabled={false}
                label="Commentaire (si pas de SIRET) : "
                maxLength={1000}
                name="comment"
                placeholder=""
                readOnly={true}
                required={true}
                rows={1}
                validate={null}
                validating={false}
              />,
            ],
            "className": "field-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "managingOffererId",
                "validator": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "disabled": false,
                "format": [Function],
                "id": null,
                "innerClassName": "col-50",
                "label": "SIRET (si applicable) : ",
                "name": "siret",
                "parse": null,
                "placeholder": "",
                "readOnly": true,
                "renderInner": [Function],
                "renderValue": [Function],
                "required": false,
                "type": "siret",
                "validate": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "disabled": false,
                "format": null,
                "id": null,
                "innerClassName": null,
                "label": "Nom : ",
                "name": "name",
                "parse": null,
                "placeholder": "",
                "readOnly": true,
                "renderInner": [Function],
                "renderValue": [Function],
                "required": true,
                "type": "text",
                "validate": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "disabled": false,
                "format": null,
                "id": null,
                "innerClassName": null,
                "label": "Nom d'usage : ",
                "name": "publicName",
                "parse": null,
                "placeholder": "",
                "readOnly": true,
                "renderInner": [Function],
                "renderValue": [Function],
                "required": false,
                "type": "text",
                "validate": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "disabled": false,
                "format": null,
                "id": null,
                "innerClassName": "col-75",
                "label": "E-mail : ",
                "name": "bookingEmail",
                "parse": null,
                "placeholder": "",
                "readOnly": true,
                "renderInner": [Function],
                "renderValue": [Function],
                "required": true,
                "type": "email",
                "validate": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "autoComplete": false,
                "className": "",
                "disabled": false,
                "label": "Commentaire (si pas de SIRET) : ",
                "maxLength": 1000,
                "name": "comment",
                "placeholder": "",
                "readOnly": true,
                "required": true,
                "rows": 1,
                "validate": null,
                "validating": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
